### PR TITLE
Fix the PIR potential attacks

### PR DESCRIPTION
**Pull request by arvid220u**

### CODE REVIEW

File: pir/fast_pir/fast_pir_client.hpp [hunk 1]
```diff
@@ -14,13 +14,21 @@
 using std::array;
 using std::bitset;
 
+#define DUMMY_INDEX 1'000'000
```
Comment (arvid220u) [inline pir/fast_pir/fast_pir_client.hpp:17]: why is dummy_index not 0 anymore? is there an attack i'm missing

File: pir/fast_pir/fast_pir_client.hpp [hunk 2]
```diff
@@ -32,6 +40,7 @@ class FastPIRClient {
   using pir_query_t =
       FastPIRQuery<seal::Serializable<seal::Ciphertext>, Galois_string>;
   using pir_answer_t = FastPIRAnswer;
+  using pir_map = std::map<pir_index_t, keys>;
 
   // TODO: fix this absolute mess of initializers???
```
Comment (arvid220u) [inline pir/fast_pir/fast_pir_client.hpp:45]: let's remove this comment. we have fixed it now :)

File: pir/fast_pir/fast_pir_client.hpp [hunk 3]
```diff
@@ -165,16 +161,19 @@ class FastPIRClient {
   }
 
  private:
+  // because we "batch" PIR encryption together, we need to know the keypair
```
Comment (arvid220u) [inline pir/fast_pir/fast_pir_client.hpp:164]: let's move these comments to the line above pir_map

**Review by arvid220u — COMMENTED** [review]
lgtm, minor comments

File: pir/fast_pir/fast_pir_client.hpp (hunk 1 above)
Comment (sualehasif) [reply pir/fast_pir/fast_pir_client.hpp:17]: basically, if you are sending to both the friend the index 0 and dummy, there is some problem. This causes overwrite bugs. — Shengtong.

File: pir/fast_pir/fast_pir_client.hpp (hunk 2 above)
Comment (sualehasif) [reply pir/fast_pir/fast_pir_client.hpp:45]: resolved.

File: pir/fast_pir/fast_pir_client.hpp (hunk 3 above)
Comment (sualehasif) [reply pir/fast_pir/fast_pir_client.hpp:164]: resolved.

**Pull request merged**